src/collidable.ts

```diff
@@ -0,0 +1,43 @@
+import { PhysObject } from './physObject';
+import { CanvasValues } from './constants';
+
+export class Collidable {
+  static willCollide(
+    object: PhysObject,
+    objects: Array<PhysObject>
+  ): Array<PhysObject> {
+    const results: Array<PhysObject> = [];
+    for (let i = 0; i < objects.length; i++) {
+      if (
+        objects[i].x + objects[i].width / 2 > object.x - object.width / 2 &&
+        objects[i].x - objects[i].width / 2 < object.x + object.width / 2 &&
+        objects[i].y + objects[i].height / 2 > object.y - object.height / 2 &&
+        objects[i].y - objects[i].height / 2 < object.y + object.height / 2
+      ) {
+        results.push(objects[i]);
+      }
+    }
+    return results;
+  }
+
+  static border(object: PhysObject) {
```

Comment:
Call this getCollidingEdges or something similar.

Reply:
I went with getCanvasEdgeCollision. I'm not fond of it but it is very explanatory.